modules/openideal_user/openideal_user.module

```diff
@@ -32,6 +33,65 @@ function openideal_user_form_user_login_form_alter(&$form, FormStateInterface $f
   ];
 }
 
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * @see _openideal_user_register_submit()
+ * @see _openideal_user_register_validate()
+ */
+function openideal_user_form_user_register_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form['first_name'] = [
+    '#type' => 'textfield',
+    '#title' => t('First name'),
+    '#required' => TRUE,
+  ];
+  $form['last_name'] = [
+    '#type' => 'textfield',
+    '#title' => t('Last name'),
+    '#required' => TRUE,
+  ];
+
+  $form['actions']['submit']['#submit'][] = '_openideal_user_register_submit';
```

Comment:
Once we add needed fields through "Manage Form Display" settings, this submit function also won't be necessary.